extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -83,7 +83,11 @@ private Transactional getTransactional(InvocationContext ic) {
     private TransactionConfiguration getTransactionConfiguration(InvocationContext ic) {
         TransactionConfiguration configuration = ic.getMethod().getAnnotation(TransactionConfiguration.class);
         if (configuration == null) {
-            return ic.getTarget().getClass().getAnnotation(TransactionConfiguration.class);
+            Object target = ic.getTarget();
```

Comment:
Shouldn't we use `getMethod().containing-class...` here then?

Reply:
Hm, yes you're right, it should work in this particular case.